src/transformations/input_output/PrintfToCoutTransformation.cpp

```diff
@@ -0,0 +1,190 @@
+#include "../../../include/transformations/input_output/PrintfToCoutTransformation.h"
+#include "../../../include/transformations/input_output/Utils.h"
+
+using clang::IfStmt, clang::CallExpr;
+using clang::Lexer, clang::CharSourceRange, clang::SourceLocation;
+using llvm::isa, llvm::cast, llvm::errs;
+using std::stoi, std::vector, std::string, std::regex_search,
+std::regex, std::sregex_iterator, std::distance;
+
+// ------------ PrintfToCoutVisitor ------------
+
+PrintfToCoutVisitor::PrintfToCoutVisitor(Rewriter * rewriter) :
+        rewriter(rewriter), sm(rewriter->getSourceMgr()), opt(rewriter->getLangOpts()) {}
+
+bool PrintfToCoutVisitor::VisitCallExpr(CallExpr *e) {
+    auto srcloc = e->getBeginLoc();
+    // Collecting printf or std::printf expressions
+    if (sm.isWrittenInMainFile(srcloc)) {
+        auto const srctext = getAsText(e);
+        if (srctext == "std::printf" || srctext == "printf") {
+            printfExpressions.push_back(e);
+        }
+    }
+    return true;
+}
+
+bool PrintfToCoutVisitor::VisitUsingDirectiveDecl(UsingDirectiveDecl *ud) {
+    auto srcloc = ud->getBeginLoc();
+    // Collecting all used namespaces
+    if (sm.isWrittenInMainFile(srcloc)) {
+        usedNamespaces.push_back(ud);
+    }
+    return true;
+}
+
+bool PrintfToCoutVisitor::determineIfSimplePrintfCommand(const CallExpr * e) {
+    string formatstring = getAsText(e->getArg(0)).str();
+
+    // We rewrite simple printf commands directly.
+    // If the number of args matches with the number from the regex,
+    // we have a relative simple printf command
+    regex const re(regexPlaceholders);
+
+    int const match_count(distance(
+            sregex_iterator(formatstring.begin(), formatstring.end(), re),
+            sregex_iterator()));
+
+    return (match_count == (e->getNumArgs() - 1));
+}
+
+void PrintfToCoutVisitor::insertHeaderAtBeginning(string headerstr) {
+    headerstr = "#include <" + headerstr + ">\n";
+    rewriter->InsertTextBefore(sm.getLocForStartOfFile(sm.getMainFileID()), headerstr);
+}
+
+StringRef PrintfToCoutVisitor::getAsText(const Expr * e) {
+    SourceLocation LocStart = e->getBeginLoc();
+    SourceLocation LocEnd = Lexer::getLocForEndOfToken(LocStart, 0, sm, opt);
+    SourceRange range(LocStart, LocEnd);
+    return Lexer::getSourceText(CharSourceRange::getCharRange(range), sm, opt);
+}
+
+string PrintfToCoutVisitor::parsePrecision(const vector<string> * matches) {
+    if (matches->size() != 1) {
+        errs() << "Format string has more then one match for \"%.Xf\"" << "\n";
+        throw MatcherException();
+    }
+    auto match = matches->at(0);
+    auto format_string = "%" + floatSplitRegex;
+    vector<string> splits = split(&match, &format_string);
```

Comment:
It's better to create a vector here and pass the pointer to it into the function.